src/Workleap.OpenApi.MSBuild/SpectralManager.cs

```diff
@@ -9,23 +8,44 @@ internal sealed class SpectralManager : ISpectralManager
 
     private readonly ILoggerWrapper _loggerWrapper;
     private readonly IHttpClientWrapper _httpClientWrapper;
-    private readonly string _toolDirectory;
+    private readonly string _spectralDirectory;
+    private readonly string _openApiToolsDirectory;
+    private readonly IProcessWrapper _processWrapper;
 
-    public SpectralManager(LoggerWrapper loggerWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper)
+    public SpectralManager(ILoggerWrapper loggerWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper, IProcessWrapper processWrapper)
     {
         this._loggerWrapper = loggerWrapper;
         this._httpClientWrapper = httpClientWrapper;
-        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, "spectral", SpectralVersion);
+        this._openApiToolsDirectory = openApiToolsDirectoryPath;
+        this._spectralDirectory = Path.Combine(openApiToolsDirectoryPath, "spectral", SpectralVersion);
+        this._processWrapper = processWrapper;
     }
 
-    public async Task InstallSpectralAsync(CancellationToken cancellationToken)
+    public async Task<string> InstallSpectralAsync(CancellationToken cancellationToken)
```

Comment:
We can have full control over how and where the file is downloaded, could we consider not returning the path and "guess" it when running the tool?

Like if the path we generate is always `openapi/swagger/<version>/spectral` or `openapi/swagger/<version>/spectral.exe`, IMO there's no need to pass it around as a parameter

Reply:
Thought about keeping this path as a private property. Wanted to get everyone's opinion on which one is a "cleaner" solution. I agree with the approach you mentioned.